spec/mongo/server/connection_pool_spec.rb

```diff
@@ -83,11 +83,21 @@
 
     context 'when min size exceeds default max size' do
       let (:options) do
-        { min_pool_size: 10 }
+        { min_pool_size: 50 }
       end
 
       it 'sets max size to equal provided min size' do
-        expect(pool.max_size).to eq(10)
+        expect(pool.max_size).to eq(50)
+      end
+    end
+
+    context 'when min size is provided and max size is zero (unlimited)' do
+      let (:options) do
+        { min_size: 10, max_size: 0 }
+      end
+
+      it 'sets max size to zero (unlimited)' do
+        expect(pool.max_size).to eq(0)
```

Comment:
There should also be an assertion that the min size is respected and unchanged from user-provided value, in this or a new test.